### PR TITLE
Cleanups and fix #233

### LINKED ISSUE #233: `win-title` does not work
```
$ /usr/bin/nsxiv -v
nsxiv v27.1-99-gbda7086
```

```
$ /usr/bin/nsxiv -T test test.jpg
nsxiv:
################################################################
#                      DEPRECATION NOTICE                      #
################################################################
# `-T` option has been deprecated in favour of `win-title`.    #
# Please read the `WINDOW TITLE` section of the manpage for    #
# more info.                                                   #
################################################################
```

```
$ cat ~/.config/nsxiv/exec/win-title
#!/bin/sh

# Example for $XDG_CONFIG_HOME/nsxiv/exec/win-title
# Called by nsxiv(1) on each redraw.
# The output is set as nsxiv's window title.
#
# Arguments, "Optional" arguments might be empty:
#   $1: resolved absolute path of the current file
#   $2: current file number
#   $3: total file number
#   $4: image width (Optional: Disabled on thumbnails mode)
#   $5: image height (Optional: Disabled on thumbnails mode)
#   $6: current zoom (Optional: Disabled on thumbnails mode)
#
# The term file is used rather than image as nsxiv does not
# precheck that the input files are valid images. Total file
# count may be different from the actual count of valid images.

exec 2>/dev/null

filename="${1##*/}"

if [ -n "$4" ]; then # image mode
	printf "%s" "nsxiv - ${filename} | ${4}x${5} ${6}% [${2}/${3}]"
else
	printf "%s" "nsxiv - ${filename} [${2}/${3}]"
fi
```

Running `/usr/bin/nsxiv test.jpg test2.jpg` opens nsxiv as expected but the window title never changes from `nsxiv`, even though I am using completely stock nsxiv configuration, including the example `win-title` script. Changing to/from thumbnail mode and switching images also does not affect the window title at all.

### PR DESCRIPTION
**Pull request by N-R-K**
* put TOP_STATUSBAR under the HAVE_LIBFONTS guard
* change get_win_title param to take unsigned char ptr
* init UTF8_STRING like other atoms
* always initialize window title
* set `WM_NAME` and `WM_ICON_NAME` inside win_set_title()

### CODE REVIEW

**Comment by N-R-K** [comment]
@TAAPArthur Is it okay to set `WM_NAME` as UTF8 encoded string? ICCCM doesn't seem to specify the encoding so it should be valid, or are there some practical downsides of this?

**Comment by defaultxr** [comment]
For what it's worth, I personally have not noticed any issues with programs that set their `WM_NAME` to a `UTF8_STRING`, though it does seem that most programs I tested (Emacs, Alacritty, Firefox) usually just set it to a regular `STRING`, at least at first. Weirdly, when I browse to a page with UTF-8 characters in the title (such as this issue) in Firefox, it changes its `WM_NAME` to a `COMPOUND_TEXT` according to `xprop`. Chromium-based browsers seem to set it to a `UTF8_STRING`. Either way, I personally do not notice any issues.

**Comment by TAAPArthur** [comment]
> @TAAPArthur Is it okay to set `WM_NAME` as UTF8 encoded string? ICCCM doesn't seem to specify the encoding so it should be valid, or are there some practical downsides of this?

I always thought it should just be a regular string. Did check the source for the [ICCCM helper libs](https://cgit.freedesktop.org/xcb/util-wm/tree/icccm/icccm.c) and they allow one to specify the encoding... st seems to set it to a UTF8 string.  So I can't think of a practical downside, but on the other hand it would be hard to notice anyways since nothing should read/use this value since we set the EWMH version. Let me confirm what happens when the client specifies a different encoding that what is set.

**Comment by N-R-K** [comment]
> it would be hard to notice anyways since nothing should read/use this value since we set the EWMH version

Yes, it's just a "fix" for certain WMs which don't support the EWMH version, see #233

We *could* just say it's "not out issue" and ask the WMs to respect EWMH version instead, but given we already have the machinery and it doesn't require much code, I don't think it'd be worth it to discriminate against such WMs.

File: main.c [hunk 1]
```diff
@@ -233,30 +233,36 @@ static bool check_timeouts(struct timeval *t)
 	return tmin > 0;
 }
 
-size_t get_win_title(char *buf, int len)
+size_t get_win_title(unsigned char *buf, int len, bool init)
```
Comment (TAAPArthur) [inline main.c:236]: It isn't immediately clear to me why we prefer unsigned char.

**Review by TAAPArthur — APPROVED** [review]

**Pull request merged**